gateway/ln-gateway/src/lib.rs

```diff
@@ -672,110 +669,110 @@ impl Gateway {
-                                            {
-                                                Ok(_) => {
-                                                    return Some(ControlFlow::<(), ()>::Continue(()))
-                                                }
-                                                Err(e) => {
-                                                    info!("Got error intercepting HTLC: {e:?}, will retry...");
-                                                }
-                                            }
-                                        } else {
-                                            info!("Got no HTLC result");
+            // Check if the HTLC corresponds to a federation supporting legacy Lightning by
+            // looking up the `Client` from the short channel id and
+            // `FederationId` (scid -> FederationId -> Client).
+            let scid_to_feds = self.scid_to_federation.read().await;
+            if let Some(short_channel_id) = htlc_request.short_channel_id {
+                let federation_id = scid_to_feds.get(&short_channel_id);
+                // Just forward the HTLC if we do not have a federation that
+                // corresponds to the short channel id
+                if let Some(federation_id) = federation_id {
+                    let clients = self.clients.read().await;
+                    let client = clients.get(federation_id);
+                    // Just forward the HTLC if we do not have a client that
+                    // corresponds to the federation id
+                    if let Some(client) = client {
+                        let cf = client
+                            .borrow()
+                            .with(|client| async {
+                                let htlc = htlc_request.clone().try_into();
+                                if let Ok(htlc) = htlc {
+                                    match client
+                                        .get_first_module::<GatewayClientModule>()
+                                        .gateway_handle_intercepted_htlc(htlc)
+                                        .await
+                                    {
+                                        Ok(_) => return Some(ControlFlow::<(), ()>::Continue(())),
+                                        Err(e) => {
+                                            info!(
+                                                "Got error intercepting HTLC: {e:?}, will retry..."
+                                            );
                                         }
-                                        None
-                                    })
-                                    .await;
-                                if let Some(ControlFlow::Continue(())) = cf {
-                                    continue;
+                                    }
+                                } else {
+                                    info!("Got no HTLC result");
```

Comment:
Can wait for a future cleanup, but we should improve these log messages as well

Reply:
I believe this section is modified in subsequent PRs so I'd like to leave it as-is for now. Probably worth doing a pass-through on the gateway code once it's refactored to clean up existing logs as well as add new ones.